code/modules/power/powernet.dm

```diff
@@ -96,4 +96,29 @@
 	if(avail >= 1000)
 		return Clamp(round(avail/10000), 10, 90) + rand(-5,5)
 	else
-		return 0
\ No newline at end of file
+		return 0
+
+/*
+compact_units:
+amount - number
+unit - text, ie.: "W", "Pa", etc.
+
+returns amount + SI prefix + unit
+up to Terra
+
+example:
+compact_units(50000,    "W") -> 50 kW
+compact_units(50000000, "W") -> 50 MW
+
+*/
+/proc/compact_units(amount, unit)
+	if(amount < 1000)
+		return "[amount] [unit]"
+	if(amount > 1000**4) // more readable than 1000000000
```

Comment:
yeah but im still pretty sure theres defines

Reply:
any idea where I should look? I tried misc.dm and math.dm in __DEFINES but nothing so far.